hnn_core/pyramidal.py

```diff
@@ -622,3 +633,56 @@ def set_biophysics(self, p_all):
                 seg.gbar_ar = 1e-6 * np.exp(3e-3 * h.distance(seg.x))
 
             h.pop_section()
+
+# The custom cell object we want to create is one with distance-dependent
+# ionic dynamics. To do so, first let's create a function for computing
+# distance-dependent ionic conductance *g*. It computes a piecewise linear
+# function where the conductance at distance ``x`` from soma is linear from
+# ``gsoma`` at x=0 (base of soma) to ``gdend`` at distance ``xkink``
+# from base of soma and constant thereafter.
+
+
+
+###############################################################################
+# Next, we need to create a subclass of the original HNN class and override
+# any methods that we would like to override. The new cell class contains
+# two new methods:
+#
+# * ``set_conductance``: sets the conductance of a particular
+#   dendritic segment of the neuron, and
+# * ``set_dends_biophys``: which loops over the segments in a cell and sets
+#   their conductance.
+
+
+class CustomL5Pyr(L5Pyr):
+    def get_g_at_dist(self, x, gsoma, gdend, xkink):
+        """Compute distance-dependent ionic conductance."""
+        if x > xkink:
+            return gdend
+        g = gsoma + x * (gdend - gsoma) / xkink
+        return g
+
+    def set_conductance(self, seg, gsoma, gdend):
+        """Insert distance dependent ionic dynamics."""
+        from neuron import h
+
+        dist_soma = abs(h.distance(seg.x))
+        seg.gbar_ca = get_g_at_dist(
+            x=dist_soma,
+            gsoma=gsoma,
+            gdend=gdend,
+            xkink=1501)  # beginning of tuft
+
+    def set_biophysics(self, p_all):
+        """Set custom dendritic biophysics"""
+        from neuron import h
+
+        L5Pyr.set_biophysics(self)
```

Comment:
What are you expecting this to do? I'm not sure...

Reply:
this is a shortcut to avoid copying everything that is `L5Pyr.set_biophysics`. It will execute all of that first and then the following lines